quixstreams/sources/manager.py

```diff
@@ -0,0 +1,241 @@
+import threading
+import logging
+import signal
+
+from pickle import PicklingError
+from typing import List
+
+from quixstreams.logging import configure_logging, LOGGER_NAME
+from quixstreams.models import Topic
+from .base import BaseSource
+from .multiprocessing import multiprocessing
+
+logger = logging.getLogger(__name__)
+
+
+class SourceProcess(multiprocessing.Process):
+    """
+    An implementation of the Source subprocess.
+
+    It manages a source and its subprocess, handles the communication between the child and parent processes,
+    lifecycle, and error handling.
+
+    Some methods are designed to be used from the parent process, and others from the child process.
+    """
+
+    def __init__(self, source):
+        super().__init__()
+        self.source: BaseSource = source
+
+        self._exceptions: List[Exception] = []
+        self._stopping = False
+
+        # copy parent process log level to the child process
+        self._loglevel = logging.getLogger(LOGGER_NAME).level
+
+        # reader and writer pipe used to communicate from the child to the parent process
+        self._rpipe, self._wpipe = multiprocessing.Pipe(duplex=False)
+
+    # --- CHILD PROCESS METHODS --- #
+
+    def _setup_signal_handlers(self):
+        """
+        Configure the child process signal handlers to handle shutdown gracefully
+        """
+        signal.signal(signal.SIGINT, self._stop)
+        signal.signal(signal.SIGTERM, self._stop)
+
+    def run(self) -> None:
+        """
+        An entrypoint of the child process.
+
+        Responsible for:
+            * Configuring the signal handlers to handle shutdown properly
+            * Execution of the source `run` method
+            * Reporting the source exceptions to the parent process
+        """
+        self._setup_signal_handlers()
+        configure_logging(self._loglevel, str(self.source), pid=True)
+        logger.info("Source started")
```

Comment:
Let's add the source name here too.
By default I see this:
```[2024-09-04 16:22:18,951] [INFO] [quixstreams] : Source started```

Reply:
Do you have any logging configuration ? I get
```
[2024-09-04 16:32:19,389] [INFO] [my_source] [165511] : Source started
```

Reply:
Nope, just a bare minimum app:
```app = Application(broker_address="localhost:9092", auto_offset_reset="earliest")```